client/core/simharness/simharness.go

```diff
@@ -0,0 +1,474 @@
+		// to be stopped here. So, it is probably fine to ignore the
+		// error returned from stop-wallet.
+		stopFn := func(ctx context.Context) {
+			<-Ctl(ctx, symbol, "./stop-wallet", rpcPort)
+		}
+		if err = ProcessCtl(ctx, symbol, stopFn, "./start-wallet", name, rpcPort, networkPort); err != nil {
+			return "", "", fmt.Errorf("%s create account error: %v", symbol, err)
+		}
+		<-time.After(time.Second * 3)
+		if symbol == "zec" {
+			<-time.After(time.Second * 10)
+		}
+		// Connect the new node to the alpha node.
+		cmdOut := <-Ctl(ctx, symbol, "./connect-alpha", rpcPort)
+		if cmdOut.Err != nil {
+			return "", "", fmt.Errorf("%s create account error: %v", symbol, cmdOut.Err)
+		}
+		<-time.After(time.Second)
+	default:
+		return "", "", fmt.Errorf("createWallet: symbol %s unknown", symbol)
+	}
+	return name, rpcPort, nil
+}
+
+var nonBtcConversionFactors = map[string]uint64{
+	"eth":       1e9,
+	"dextt.eth": 1e9,
+}
+
+func conversionFactor(symbol string) uint64 {
+	if c := nonBtcConversionFactors[symbol]; c > 0 {
+		return c
+	}
+	return 1e8
+}
+
+// ValString returns a string representation of the value in conventional
+// units.
+func ValString(v uint64, assetSymbol string) string {
+	precisionStr := fmt.Sprintf("%%.%vf", math.Log10(float64(conversionFactor(assetSymbol))))
+	return fmt.Sprintf(precisionStr, float64(v)/float64(conversionFactor(assetSymbol)))
+}
+
+// FindOpenAddrs finds unused addresses.
+func FindOpenAddrs(n int) ([]net.Addr, error) {
+	addrs := make([]net.Addr, 0, n)
+	for i := 0; i < n; i++ {
+		addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+		if err != nil {
+			return nil, err
+		}
+
+		l, err := net.ListenTCP("tcp", addr)
+		if err != nil {
+			return nil, err
+		}
+		defer l.Close()
+		addrs = append(addrs, l.Addr())
+	}
+
+	return addrs, nil
+}
+
+// process stores a long running command and the funcion to stop it on shutdown.
+type process struct {
+	cmd    *exec.Cmd
+	stopFn func(ctx context.Context)
+}
+
+var (
+	processesMtx sync.Mutex
+	processes    []*process
+)
+
+// ProcessCtl will run the long running command from the harness-ctl
+// directory for the specified symbol. The command and stop function are saved
+// to a global slice for stopping later.
+func ProcessCtl(ctx context.Context, symbol string, stopFn func(context.Context), cmd string, args ...string) error {
+	processesMtx.Lock()
+	defer processesMtx.Unlock()
+	if err := ctx.Err(); err != nil {
+		return err
+	}
+	dir := filepath.Join(Dir, symbol, "harness-ctl")
+	// Killing the process with ctx or process.Kill *sometimes* does not
+	// seem to stop the coin daemon. Kill later with an rpc "stop" command
+	// contained in the stop function.
+	command := exec.Command(cmd, args...)
+	command.Dir = dir
+	p := &process{
+		cmd:    command,
+		stopFn: stopFn,
+	}
+	if err := command.Start(); err != nil {
+		return fmt.Errorf("unable to start process: %v", err)
+	}
+	processes = append(processes, p)
+	return nil
+}
```

Comment:
Some fairly significant duplication from loadbot.  More trouble than it's worth to find a common home for these kinds of harness interactions?